libp2p/protocols/identify.nim

```diff
@@ -101,16 +106,17 @@ proc decodeMsg*(buf: seq[byte]): Option[IdentifyInfo] =
     trace "decodeMsg: failed to decode received message"
     none[IdentifyInfo]()
 
-proc newIdentify*(peerInfo: PeerInfo): Identify =
+proc newIdentify*(peerInfo: PeerInfo, provider: CurrentAddressProvider = nil): Identify =
   new result
   result.peerInfo = peerInfo
+  result.provider = provider
   result.init()
 
 method init*(p: Identify) =
   proc handle(conn: Connection, proto: string) {.async, gcsafe, closure.} =
     try:
       trace "handling identify request", conn
```

Comment:
I think there is a misunderstanding as to what `observedAddr` does - it stores the address that the remote node has seen us connect from. It doesn't make sense to sed it here, because we essentially sending back what the remote sent us.

Reply:
Yeah I realized that after, should be addrs. I rationalized in the issue about it.